Deployment handler panic: nil pointer dereference when event.Namespace is empty (SQS events)

## Bug Report

### Description

When using a single Config with both `AwsSqs` and `KubernetesSecret` notification sources and `Deployment` + `ExternalSecret` destinations, the Reloader panics with a nil pointer dereference when processing SQS events. This is because SQS events have an empty `Namespace` field, which causes the Deployment handler's `Filter` function to pass a nil `client.ListOption` to `client.List`.

### Version

- Reloader: v0.0.5

### Config

```yaml
apiVersion: reloader.external-secrets.io/v1alpha1
kind: Config
metadata:
  name: eso-reloader
spec:
  notificationSources:
    - type: AwsSqs
      awsSqs:
        queueURL: https://sqs.us-east-1.amazonaws.com/xxx/infra-eso-reloader-secrets-events
        auth:
          authMethod: irsa
          region: us-east-1
          serviceAccountRef:
            name: reloader-controller-manager
            namespace: infra-eso-reloader
    - type: KubernetesSecret
      kubernetesSecret:
        serverURL: https://kubernetes.default.svc
        labelSelector:
          matchLabels:
            lemonade.com/managed-by: external-secrets
  destinationsToWatch:
    - type: ExternalSecret
      externalSecret:
        labelSelectors:
          matchLabels: {}
    - type: Deployment
      deployment:
        labelSelectors:
          matchLabels: {}
```

### Panic Stack Trace

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x16552e0]

goroutine 248 [running]:
sigs.k8s.io/controller-runtime/pkg/client.(*ListOptions).ApplyOptions(...)
    /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.21.0/pkg/client/options.go:560
sigs.k8s.io/controller-runtime/pkg/cache/internal.(*CacheReader).List(...)
    /home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.21.0/pkg/cache/internal/cache_reader.go:113 +0x80
...
github.com/external-secrets-inc/reloader/internal/handler/deployment.(*Handler).Filter(...)
    /home/runner/work/reloader/reloader/internal/handler/deployment/handler.go:41 +0x126
github.com/external-secrets-inc/reloader/internal/handler.(*EventHandler).HandleEvent(...)
    /home/runner/work/reloader/reloader/internal/handler/processor.go:50 +0x428
```

### Root Cause

In `internal/handler/deployment/handler.go`, the `Filter` function:

```go
var opt client.ListOption
if event.Namespace != "" {
    opt = client.InNamespace(event.Namespace)
}
if err := h.client.List(h.ctx, &deployments, opt); err != nil {
```

When `event.Namespace` is empty (which is always the case for SQS events), `opt` remains nil. Passing a nil `client.ListOption` to `client.List` causes a nil pointer dereference in `ApplyOptions`.

### Suggested Fix

```go
var opts []client.ListOption
if event.Namespace != "" {
    opts = append(opts, client.InNamespace(event.Namespace))
}
if err := h.client.List(h.ctx, &deployments, opts...); err != nil {
```

### Additional Finding: Shared EventHandler across Configs

While debugging this, we also discovered that when using **multiple Config resources**, the EventHandler's destination cache is overwritten by the last Config to reconcile. This means only the last Config's `destinationsToWatch` are ever processed, regardless of which notification source triggers the event.

This forced us to consolidate into a single Config (which then exposed this nil pointer bug). Multiple Configs with different destinations don't work as expected.

### Steps to Reproduce

1. Create a Config with `AwsSqs` notification source and both `ExternalSecret` + `Deployment` destinations
2. Trigger a secret change in AWS Secrets Manager
3. SQS message is received, event is published with `Namespace: ""`
4. Deployment handler's `Filter` is called → panic

### Workaround

Use only `KubernetesSecret` notification source (which always provides a Namespace). KubernetesSecret events work correctly with the Deployment handler.

--- a/internal/handler/deployment/handler.go
+++ b/internal/handler/deployment/handler.go
@@ -34,11 +34,11 @@ func (h *Handler) Filter(destination *v1alpha1.DestinationToWatch, event events.
 	}
 	logger := log.FromContext(h.ctx)
 	var deployments appsv1.DeploymentList
-	var opt client.ListOption
+	var opts []client.ListOption
 	if event.Namespace != "" {
-		opt = client.InNamespace(event.Namespace)
+		opts = append(opts, client.InNamespace(event.Namespace))
 	}
-	if err := h.client.List(h.ctx, &deployments, opt); err != nil {
+	if err := h.client.List(h.ctx, &deployments, opts...); err != nil {
 		return nil, fmt.Errorf("failed to list Deployments:%w", err)
 	}
 	for key, deployment := range deployments.Items {

